test/test_utils/ec2.py

```diff
@@ -509,6 +513,46 @@ def enforce_IMDSv2(instance_id, region=DEFAULT_REGION, ec2_client=None, hop_limi
     LOGGER.info(f"Modify Metadata options State of EC2 instance: {state}")
 
 
+
+def enforce_IMDSv1(instance_id, region=DEFAULT_REGION, ec2_client=None):
+    """
+    Enabled IMDSv1 on EC2 instance.
+
+    :param instance_id: str, ec2 instance id
+    :param region: str, Region where ec2 instance is launched.
+    :param ec2_client: str, ec2 client.
+    :param hop_limit: str, hop limit to be set on ec2 instance.
+    """
+    ec2_client = ec2_client or get_ec2_client(region)
+    response = ec2_client.modify_instance_metadata_options(
+        InstanceId=instance_id,
+        HttpTokens='optional',
+    )
+
+    if not response:
+        raise Exception("Unable to enforce IMDSv1. No response received.")
+
+    timeout = 3
+    state = None
+    if response["InstanceId"]:
+        while timeout > 0:
+            time.sleep(timeout)
+            LOGGER.info(f"Slept for: {timeout}")
+            res = ec2_client.describe_instances(InstanceIds=[instance_id])
+            if res:
+                metadata_options = res['Reservations'][0]['Instances'][0]['MetadataOptions']
+                http_tokens = metadata_options['HttpTokens']
+                state = metadata_options['State']
+
+                if http_tokens == 'optional' and state == 'applied':
+                    break
+            timeout -= 1
+
+    if state == 'pending' or timeout == 0:
```

Comment:
Wouldn't it better to check if `state != "applied"` here?

Reply:
Yes